In [3]:
import DALI as dali_code
from langdetect import detect, DetectorFactory
from tqdm import tqdm
import pandas as pd
import os

DetectorFactory.seed = 20250101  # fix seed for reproducibility

In [ ]:
dali_data_path = '../../data/DALI/DALI_v2.0/annot_tismir'
dali_data = dali_code.get_the_DALI_dataset(dali_data_path, skip=[], keep=[])
song_ids = list(dali_data.keys())

English_songs = []
for id in tqdm(song_ids):
    entry = dali_data[id]
    entry_meta = entry.info['metadata']

    paras = entry.annotations['annot']['paragraphs']

    song_lyrics = []
    for para in paras:
        song_lyrics.append(para['text'])
    song_lyrics = ' '.join(song_lyrics)

    languages_detected = detect(song_lyrics)
    
    if entry_meta['language'] == 'english' and languages_detected == 'en':
        English_songs.append(id)

print(f"Total English songs: {len(English_songs)}")

/home/ids/chawang/miniconda3/envs/mlm_new/lib/python3.12/site-packages/DALI/utilities.py:110: VisibleDeprecationWarning: dtype(): align should be passed as Python or NumPy boolean but got `align=0`. Did you mean to pass a tuple to create a subarray type? (Deprecated NumPy 2.4)
  output = pickle.load(f)
100%|██████████| 7756/7756 [00:21<00:00, 360.48it/s]

Total English songs: 5814


In [ ]:
df = pd.read_csv('../../data/DALI_English_songs.csv')
df

,English_song_ids,split
0,00070c7c333849e4a3725b906c339042,development
1,000b837fdd034df58df3f511f3c49d5e,development
2,000cba3f7b854ace8c9bfa162621f581,development
3,001560cf57384035b3f841bcf75fcbb9,development
4,001651b1987148ea96f964df906186d9,development
...,...,...
5809,ffe52f06434340e182e3714a676ba6fd,development
5810,ffebd58e2ea344ad8e7f1fea3f974373,development
5811,ffecb6f72fca4c51bc649bd61d7264de,development
5812,ffef85420e914f96bad1420f75a9b105,development


Note: There are 38 songs with misaligned melody and lyrics lines, which are detected during feature extraction. These songs therefore were not included, and the valid number of songs are 5814 - 38 = 5776.